Composer/packages/client/src/recoilModel/atoms/appState.ts

```diff
@@ -364,3 +364,8 @@ export const warnAboutFunctionsState = atom<boolean>({
   key: getFullyQualifiedKey('warnAboutFunctionsState'),
   default: false,
 });
+
+export const localTemplatePathState = atom<string>({
```

Comment:
This is a temporary state that its life is bound to the creation view, why does it need to be an atom instead of a local component state that gets passed around as prop?

Reply:
I generally lean towards central state even if it is bound to the creation view as it results in less files touched with prop passing and consistency. However you are right in that most of Composer code does favor local state in these scenarios so I have snapped to that standard.

Reply:
The rule of thumb is that UI state belongs to the components and app state belongs to the recoil atoms. UI state gets garbage collected when the component goes away from screen, while app state hangs around for the life of the app, which in this case is totally unnecessary. You can easily imagine that every component brings its UI state to the recoil atoms and now we have memory leaks all over.